cclib/method/moments.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2018, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Calculation of electric multipole moments based on data parsed by cclib."""
+
+import numpy
+
+from cclib.parser.utils import convertor
+from cclib.method.calculationmethod import Method
+
+
+class Moments(Method):
+    def __init__(self, data):
+        self.required_attrs = ('atomcoords', 'atomcharges')
+
+        super(Moments, self).__init__(data)
+
+    def __str__(self):
+        """Returns a string representation of the object."""
+        return "Multipole moments of %s" % (self.data)
+
+    def __repr__(self):
+        """Returns a representation of the object."""
+        return 'Moments("%s")' % (self.data)
+
+    def _calculate_dipole(self, charges, coords, origin):
+        coords_au = convertor(coords, 'Angstrom', 'bohr')
+        
+        if self.data.charge == 0:
+            dipole = numpy.dot(charges, coords_au)
+        else:
+            origin_au = convertor(origin, 'Angstrom', 'bohr')
+            dipole = numpy.dot(charges, coords_au - origin_au)
+            
+        return convertor(dipole, 'ebohr', 'Debye')
+
+    def _calculate_quadrupole(self, charges, coords, origin):
+        transl_coords_au = convertor(coords - origin, 'Angstrom', 'bohr')
+
+        delta = numpy.eye(3)
+        Q = numpy.zeros([3,3])
+        for i in range(3):
+            for j in range(3):
+                for q, r in zip(charges, transl_coords_au):
+                    Q[i,j] += q * (3 * r[i] * r[j] - \
+                              numpy.linalg.norm(r)**2 * delta[i][j])
+
+        triu_idxs = numpy.triu_indices_from(Q)
+        raveled_idxs = numpy.ravel_multi_index(triu_idxs, Q.shape)
+        quadrupole = numpy.take(Q.flatten(), raveled_idxs)
+        
+        return convertor(quadrupole, 'ebohr2', 'Buckingham')
+    
+    def calculate(self, origin='nuccharge', population='mulliken',
+                  masses=None, overwrite=False):
```

Comment:
`overwrite` is also not a good name here since the output file might not contain multipole moments. Maybe `update` (a data object) or `save` (results) is much better.

Reply:
I like `save`, the reason being that other methods (like MPA) store their results as attributes in the `Method` instance, not the `ccData` instance.

Reply:
In this case, it seems there is no need for `save` argument.